package.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  pkgs,
+  pkg-config,
+  rustPlatform,
+  makeWrapper,
+}:
+let
+  meta = (builtins.fromTOML ./Cargo.toml).package;
+in
+rustPlatform.buildRustPackage rec {
+  name = meta.name;
+  version = meta.version;
+
+  src = lib.cleanSource ./.;
+
+  cargoHash = "sha256-LiH7OqQL7te1GVF3qfYVRQpAhQmsVGtgwhKhZorQp2k=";
```

Comment:
Could you please put the cargoHash in the function args at the top, for easier overriding if it ever needs to be overriden?

Reply:
I can do that... How it be used?

Reply:
Well, if the Cargo.toml changes in the repo, the cargoHash will need to be changed. Instead of waiting or fiddling around with overrideAttrs (which might not be possible), someone could just do `pkgs.hyprswitch.override { cargoHash = "newhash"; }`

Reply:
That's definitely the way, forgot it existed

Reply:
@justDeeevin I think you can just remove the cargoHash and use ``cargoLock = {lockFile = ./Cargo.lock;};`` instead; 
I also managed to set up a NixOs Testmachine to test this